### PR TITLE
Use case for self signed ca cert

### PR DESCRIPTION
**Pull request by shdeshpa07**
## Jira 

* [OADP-4918](https://issues.redhat.com/browse/OADP-4918)

Added a use case enabling a self-signed CA cert

##  Version

* OCP 4.13 → OCP 4.18

## Preview

* [Self-signed CA cert use case](https://84466--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-use-cases/oadp-usecase-enable-ca-cert.html)

## QE Review

* [x] QE has approved this change.

### CODE REVIEW

**Comment by shdeshpa07** [comment]
@anarnold97 - Please review the PR. Thanks.

File: backup_and_restore/application_backup_and_restore/oadp-use-cases/oadp-usecase-enable-ca-cert.adoc
Comment (anarnold97) [inline backup_and_restore/application_backup_and_restore/oadp-use-cases/oadp-usecase-enable-ca-cert.adoc:9]:
```suggestion
The following use case provides the steps for including a self-signed CA certificate in the Data Protection Application (DPA) and then running a backup. You store the backup in a NooBaa bucket provided by {odf-first}.
```
sometimes i hate myself

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (anarnold97) [inline modules/oadp-usecase-include-ca-cert-backup.adoc:9]:
```suggestion
The `s3.openshift-storage.svc` service, provided by {odf-short}, uses a TLS certificate that is signed with the self-signed service CA. To prevent a `certificate signed by unknown authority` error, you must include a self-signed CA certificate in the backup storage location (BSL) section of the Data Protection Application (DPA). In this use case, you:
```

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (anarnold97) [inline modules/oadp-usecase-include-ca-cert-backup.adoc:13]:
```suggestion
* Include a self-signed CA certificate in the DPA
```
you ding-dong! LOL

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (anarnold97) [inline modules/oadp-usecase-include-ca-cert-backup.adoc:14]:
```suggestion
* Run a backup of an application
```

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (anarnold97) [inline modules/oadp-usecase-include-ca-cert-backup.adoc:173]:
```suggestion
. Verify that the DPA has been created successfully by running the following command:In the example output, you can see the `status` object has the `type` field set to `Reconciled`. This means the DPA has been successfully created.
```

Very personal and pedantic but i think the present perfect continuous tense of `has been` reads better, which might be wrong

**Comment by shdeshpa07** [comment]
@PrasadJoshi12  @weshayutin  - PR is ready for your review please. Thanks.

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (PrasadJoshi12) [inline modules/oadp-usecase-include-ca-cert-backup.adoc:164]:
```suggestion
<3> Copy and paste the `Base64` encoded certificate from an earlier step.
```

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (PrasadJoshi12) [inline modules/oadp-usecase-include-ca-cert-backup.adoc:112]: BSL goes into unavailable status with service-ca.crt. It works with ca.crt
```suggestion
  -o jsonpath='{.data.ca\.crt}' | base64 -w0; echo
```

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (PrasadJoshi12) [inline modules/oadp-usecase-include-ca-cert-backup.adoc:164]:
```suggestion
<3> Copy and paste the `Base64` decoded certificate from an earlier step.
```

**Review by PrasadJoshi12 — APPROVED** [review]

**Comment by dfitzmau** [comment]
Hi @shdeshpa07 . The [2 commits](https://docs.google.com/document/d/1fLMpK4bqthtFlCwA36yeo2cI-pExFpX76VMnSi_In6Q/edit?pli=1&tab=t.0#heading=h.epes4xf06pzk) need to be squashed into 1 commit.

File: backup_and_restore/application_backup_and_restore/oadp-use-cases/oadp-usecase-enable-ca-cert.adoc
Comment (dfitzmau) [inline backup_and_restore/application_backup_and_restore/oadp-use-cases/oadp-usecase-enable-ca-cert.adoc:9]:
```suggestion
The following use case provides the steps for including a self-signed Certificate Authority (CA) certificate in the Data Protection Application (DPA) and then running a backup. You store the backup in a NooBaa bucket provided by {odf-first}.
```

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (dfitzmau) [inline modules/oadp-usecase-include-ca-cert-backup.adoc:7]:
```suggestion
= Backing up an application and its self-signed CA certificate
```
The "including" dangles a bit, but adding more text would mean a longer title, which is not ideal.

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (dfitzmau) [inline modules/oadp-usecase-include-ca-cert-backup.adoc:9]:
```suggestion
The `s3.openshift-storage.svc` service, provided by {odf-short}, uses a Transport Layer Security protocol (TLS) certificate that is signed with the self-signed service CA. 

To prevent a `certificate signed by unknown authority` error, you must include a self-signed CA certificate in the backup storage location (BSL) section of the Data Protection Application (DPA). For this situation, you must complete the following tasks:
```

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (dfitzmau) [inline modules/oadp-usecase-include-ca-cert-backup.adoc:11]: These all seem like full sentences so please add a full stop to each bulleted sentence.

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (dfitzmau) [inline modules/oadp-usecase-include-ca-cert-backup.adoc:107]:
```suggestion
. Extract the service CA certificate from the `openshift-service-ca.crt` config map by running the following command. Ensure that you encode the certificate in `Base64` format and `echo` the value to use in a later step.
```

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (dfitzmau) [inline modules/oadp-usecase-include-ca-cert-backup.adoc:123]:
```suggestion
. Configure the Data Protection Application (DPA) with the bucket name and CA certificate as shown in the following example:
```

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (dfitzmau) [inline modules/oadp-usecase-include-ca-cert-backup.adoc:162]:
```suggestion
<1> The `insecureSkipTLSVerify` flag can be set to either `true` or `false`. If set to "true", SSL/TLS security is disabled. If set to `false`, SSL/TLS security is enabled.
```

Comment (dfitzmau) [reply modules/oadp-usecase-include-ca-cert-backup.adoc:162]: See "Use backticks for all non-GUI "system items", including:" in this [guide](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference).

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (dfitzmau) [inline modules/oadp-usecase-include-ca-cert-backup.adoc:173]: Consider a `.Verification` heading to offset the verification steps from the procedure. See https://redhat-documentation.github.io/modular-docs/#verification

If the flow of steps cannot be changed, please ignore my comment.

**Comment by dfitzmau** [comment]
Hi @shdeshpa07 . Great work! I've finished my review and added comments inline. Please contact me if you have any queries. The 2 commits will need to be squashed into 1 commit.

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (shdeshpa07) [reply modules/oadp-usecase-include-ca-cert-backup.adoc:175]: This step is embedded in the procedure and has been added for users to get in the habit of checking the DPA status. It does break the flow if I separate it out. Thanks.

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (shdeshpa07) [reply modules/oadp-usecase-include-ca-cert-backup.adoc]: Done. Thanks.

**Review by xenolinux — COMMENTED** [review]
Before merging I have couple of suggestions and comments. 

Let me know if you have any questions. Good work!!

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (shdeshpa07) [reply modules/oadp-usecase-include-ca-cert-backup.adoc:125]: You have a point here, but in OADP elsewhere, we have not followed this pattern. For the sake of consistency, I am not changing it here. Thanks for the suggestion though!

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (shdeshpa07) [reply modules/oadp-usecase-include-ca-cert-backup.adoc:168]: Since I am not following the suggestion to create a yaml file, 'Create the DPA' makes more sense and is consistent with other doc sets. Thanks for suggestion.

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (shdeshpa07) [reply modules/oadp-usecase-include-ca-cert-backup.adoc:207]: It has been flagged to me in reviews for other PRs to keep the top part of the CR so that the user can identify what is the type of the CR (though is clearly obvious that is an example output). So, keeping it as is. Thanks.

Comment (xenolinux) [reply modules/oadp-usecase-include-ca-cert-backup.adoc:207]: No worries. The top part is required essentially in the step's YAML where we define the CR. In the example output, you can avoid long outputs to focus on the expected output part.

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (shdeshpa07) [reply modules/oadp-usecase-include-ca-cert-backup.adoc:11]: The BSL is the object in the DPA.

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (shdeshpa07) [reply modules/oadp-usecase-include-ca-cert-backup.adoc:57]: done.

File: modules/oadp-usecase-include-ca-cert-backup.adoc
Comment (shdeshpa07) [reply modules/oadp-usecase-include-ca-cert-backup.adoc:11]: changed to `DataProtectionApplication` CR at all places. thanks.

**Comment by shdeshpa07** [comment]
> Before merging I have couple of suggestions and comments.
> 
> Let me know if you have any questions. Good work!!

Thank you Servesha for the review. These are indeed very good suggestions. Apart from just a couple of suggestions, I have included all of your review comments. Just an observation though, the reviews vary from reviewer to reviewer. From my experience, some reviewers  are very strong on avoiding 'can'. Some, not so much. Sometimes, it gets really confusing whether to write in a particular way or not :). But I appreciate your thoroughness and the effort in reviewing the content. Thanks again!

**Comment by xenolinux** [comment]
> > Before merging I have couple of suggestions and comments.
> > Let me know if you have any questions. Good work!!
> 
> Thank you Servesha for the review. These are indeed very good suggestions. Apart from just a couple of suggestions, I have included all of your review comments. Just an observation though, the reviews vary from reviewer to reviewer. From my experience, some reviewers are very strong on avoiding 'can'. Some, not so much. Sometimes, it gets really confusing whether to write in a particular way or not :). But I appreciate your thoroughness and the effort in reviewing the content. Thanks again!

I understand. No worries :)

This can be merged once commits are squashed and builds pass. I will keep an eye on the build checks.

**Comment by shdeshpa07** [comment]
> > > Before merging I have couple of suggestions and comments.
> > > Let me know if you have any questions. Good work!!
> > 
> > 
> > Thank you Servesha for the review. These are indeed very good suggestions. Apart from just a couple of suggestions, I have included all of your review comments. Just an observation though, the reviews vary from reviewer to reviewer. From my experience, some reviewers are very strong on avoiding 'can'. Some, not so much. Sometimes, it gets really confusing whether to write in a particular way or not :). But I appreciate your thoroughness and the effort in reviewing the content. Thanks again!
> 
> I understand. No worries :)
> 
> This can be merged once commits are squashed and builds pass. I will keep an eye on the build checks.

Cool. Thank you :). I just squashed the commits. Will take a bit for the build to pass.

**Comment by shdeshpa07** [comment]
> > > > Before merging I have couple of suggestions and comments.
> > > > Let me know if you have any questions. Good work!!
> > > 
> > > 
> > > Thank you Servesha for the review. These are indeed very good suggestions. Apart from just a couple of suggestions, I have included all of your review comments. Just an observation though, the reviews vary from reviewer to reviewer. From my experience, some reviewers are very strong on avoiding 'can'. Some, not so much. Sometimes, it gets really confusing whether to write in a particular way or not :). But I appreciate your thoroughness and the effort in reviewing the content. Thanks again!
> > 
> > 
> > I understand. No worries :)
> > This can be merged once commits are squashed and builds pass. I will keep an eye on the build checks.
> 
> Cool. Thank you :). I just squashed the commits. Will take a bit for the build to pass.

Hi Servesha, PR is ready for merge please :) Thanks.

**Pull request merged**
